lib/ui/hooks.dart

```diff
@@ -53,8 +53,9 @@ void _updateLocales(List<String> locales) {
   final int numLocales = locales.length ~/ stringsPerLocale;
   window._locales = new List<Locale>(numLocales);
   for (int localeIndex = 0; localeIndex < numLocales; localeIndex++) {
-    window._locales[localeIndex] = new Locale(locales[localeIndex * stringsPerLocale],
-                                   locales[localeIndex * stringsPerLocale + 1]);
+    window._locales[localeIndex] = new Locale.fromSubtags( languageCode: locales[localeIndex * stringsPerLocale],
```

Comment:
Conventional Flutter formatting would be:
```
window._locales[localeIndex] = new Locale.fromSubtags(
  languageCode: locales[localeIndex * stringsPerLocale],
  countryCode: locales[localeIndex * stringsPerLocale + 1] == '' ? null : locales[localeIndex * stringsPerLocale + 1],
  scriptCode: locales[localeIndex * stringsPerLocale + 2] == '' ? null : locales[localeIndex * stringsPerLocale + 2],
);
```
It would be a little clearer if the strings lookups were factored out. And `isEmpty`.
```
final String countryCode = locales[localeIndex * stringsPerLocale + 1];
final String scriptCode = locales[localeIndex * stringsPerLocale + 2];

window._locales[localeIndex] = new Locale.fromSubtags(
  languageCode: locales[localeIndex * stringsPerLocale],
  countryCode: countryCode.isEmpty ? null : countryCode,
  scriptCode: scriptCode.isEmpty ? null : scriptCode;
);
```

Should the languageCode value be validated? If langaugeCode.trim().isEmpty, maybe skip the locale.